internal/core/data/v2/application/event.go

```diff
@@ -158,6 +158,20 @@ func EventCountByDevice(deviceName string, dic *di.Container) (uint32, errors.Ed
 	return count, nil
 }
 
+// The DeletePushedEvents function will be invoked by controller functions
+// and then invokes DeletePushedEvents function in the infrastructure layer to remove
+// all events that have been pushed (pushed timestamp is greater than zero)
+func DeletePushedEvents(dic *di.Container) errors.EdgeX {
+	dbClient := v2DataContainer.DBClientFrom(dic.Get)
+
+	err := dbClient.DeletePushedEvents()
```

Comment:
This func should have the logic to get the ids of pushed events and send to async channel.